src/DataTypeRegistry.php

```diff
@@ -193,6 +193,17 @@ public function isRegistered( $typeId ) {
 		return isset( $this->typeDataItemIds[$typeId] );
 	}
 
+	/**
+	 * @since 3.2
+	 *
+	 * @param string $typeId
+	 *
+	 * @return boolean
+	 */
+	public function isRecordType( string $typeId ) : bool {
+		return strpos( $typeId, '_rec' ) !== false;
+	}
```

Comment:
Hmm... ideally there would be a dedicated type (class/interface) for the data types. Just a semi related idea, not something for this PR